broker/api.py

```diff
@@ -278,59 +263,35 @@ def update(
+                if params["origin"]:
+                    origin_hostname = params["origin"]
                 else:
-                    instance.cloudfront_origin_hostname = params["origin"]
+                    origin_hostname = config.DEFAULT_CLOUDFRONT_ORIGIN
+                instance.cloudfront_origin_hostname = origin_hostname
 
             if "path" in params:
-                if params["path"] is None:
-                    instance.cloudfront_origin_path = ""
+                if params["path"]:
+                    cloudfront_origin_path = params["path"]
                 else:
-                    instance.cloudfront_origin_path = params["path"]
+                    cloudfront_origin_path = ""
+                instance.cloudfront_origin_path = cloudfront_origin_path
             if "forward_cookies" in params:
-                forward_cookies = params["forward_cookies"]
-                if forward_cookies is None:
-                    forward_cookies = "*"
-
-                forward_cookies = forward_cookies.replace(" ", "")
-                if forward_cookies == "":
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.NONE.value
-                    )
-                    instance.forwarded_cookies = []
-                elif forward_cookies == "*":
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.ALL.value
-                    )
-                    instance.forwarded_cookies = []
-                else:
-                    instance.forward_cookie_policy = (
-                        CDNServiceInstance.ForwardCookiePolicy.WHITELIST.value
-                    )
-                    instance.forwarded_cookies = forward_cookies.split(",")
+                forward_cookie_policy, forwarded_cookies = parse_cookie_options(params)
+                instance.forward_cookie_policy = forward_cookie_policy
+                instance.forwarded_cookies = forwarded_cookies
 
             if "forward_headers" in params:
-                forwarded_headers = params["forward_headers"]
-                if not forwarded_headers:
-                    forwarded_headers = []
-                else:
-                    forwarded_headers = forwarded_headers.replace(" ", "")
-                    forwarded_headers = forwarded_headers.split(",")
-                if (
-                    instance.cloudfront_origin_hostname
-                    == config.DEFAULT_CLOUDFRONT_ORIGIN
-                ):
-                    forwarded_headers.append("HOST")
-                instance.forwarded_headers = forwarded_headers
+                forwarded_headers = parse_header_options(params)
+            else:
+                # this is a weird way to do this, but if we _don't_ do this, sqlalchemy doesn't
+                # recognize that instance.forwarded_headers gets changed
+                forwarded_headers = [*instance.forwarded_headers]
```

Comment:
I really want a more readable thing here. The best I can come up with is something like:
```python
def copy_but_dont_tell_sqlalchemy(list_):
    new_list = [*list_]
    return new_list
```